keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -395,15 +396,36 @@ class SqlArtifactRepository(
         .execute()
     }
     sqlRetry.withRetry(WRITE) {
+      // update old "CURRENT" to "PREVIOUS, set promotionReference for use in summary data
       jooq
         .update(ENVIRONMENT_ARTIFACT_VERSIONS)
         .set(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS, PREVIOUS.name)
+        .set(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_REFERENCE, version)
         .where(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(environmentUid))
         .and(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artifact.uid))
         .and(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS.eq(CURRENT.name))
         .and(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.ne(version))
         .execute()
     }
+    sqlRetry.withRetry(WRITE) {
```

Comment:
Should wrap these in a transaction now that there's more than one write operation.